### PR TITLE
Crossvalidation

### PR DESCRIPTION
**Pull request by pfistfl**

### CODE REVIEW

**Comment by pfistfl** [comment]
Updated with @Arshadipour 's rfc

**Comment by SteveBronder** [comment]
Dunno who is working on this, but another thing to keep in mind is that if your index has multiple values on the same date make sure all those days are in either the test or training set!

**Comment by pfistfl** [comment]
Hey @SteveBronder 
Just out of interest:
What for example about time-series that go over only one day and are for example measured every minute?
We might need to really define what a time-series is.

**Comment by SteveBronder** [comment]
> What for example about time-series that go over only one day and are for example measured every minute?
> We might need to really define what a time-series is.

Same thing applies, think about the below table

Index | Minute | Stock | Value |
|------|------|-------|-------|
| 1 | 1:00    | APPL  | 1     |
| 2| 1:00    | GOOG  | 10    |
| 3| 2:00    | APPL  | 1     |
| 4| 2:00    | GOOG  | 10    |
| 5| 3:00    | APPL  | 100   |
| 6| 3:00    | GOOG  | 1000  |
| 7| 4:00    | APPL  | 100   |

Imagine you cut for train as index 1:5 and test on 6:7. Your leaking information from the test set into the training set

Zak David wrote a good bit one this below

https://content.iospress.com/articles/algorithmic-finance/af190900


eod make sure you are preserving the time component the user tells you. I've been jinxed more than once from data collected in a different time zone causing leakage
